File: YukariLauncher/src/main/java/net/kdt/pojavlaunch/authenticator/listener/DoneListener.java
package net.kdt.pojavlaunch.authenticator.listener;

import net.kdt.pojavlaunch.value.MinecraftAccount;

/**
 * ログインが完了しアカウントが受信されたときに呼び出されます。UIスレッドで実行されることが保証されています。
 */
public interface DoneListener {
    /**
     * ログインが完了したときに呼び出されます。
     * @param account ログインしたアカウント
     */
    void onLoginDone(MinecraftAccount account);
}

File: YukariLauncher/src/main/java/com/arata/yukarilauncher/feature/login/DoneListener.kt
package com.arata.yukarilauncher.feature.login

import com.arata.yukarilauncher.value.MinecraftAccount

/** ログイン完了リスナー。 */
interface DoneListener {
    /** ログインが完了したときに呼び出される。 @param account ログインアカウント */
    fun onLoginDone(account: MinecraftAccount)
}
